Add RAC styles

p-galligan opened the pull request:
Fixes #252 
Implements the RAC Style Library v0.12.0.

Also implements BackstopJS for future visual regression testing.

p-galligan committed:
Add initial changes

Work through footer in initial meeting.

daremiyo committed:
Style edits to header
- Text brand and nav items
- Takes out old styling

p-galligan committed:
tweak homepage

Tweaks homepage styles. Not done with changes, but want to push these up.

helrond reviewed (commented):
Looks like you've actually made a fair amount of progress here!

I think there's probably going to be a lot of work to do once you disable the `main.css` file in the `head.html` include. But I might advocate for doing that sooner rather than later so you don't have to create custom rules twice. 

The other thing I'm thinking about, because this is a blog, is really making readability as good as possible. I think our style library is okay in that regards but I think we can make a case for cranking up the font size a bit, and making sure we've got line length, spacing, kerning and all that good stuff right.

daremiyo commented:
Work done so far:
Focused on homepage
- Implemented components for footer and header
- Made some tweaks to components based on what was specific to blog site
- Some edits to make the new components fit better

Work to do/questions:
- Typography
- Figuring out how to get logo into homepage
- Closer look at pages other than homepage

p-galligan committed:
Comment out main css

Comments out the main.css from the head so we can tackle layout from the ground up.

p-galligan committed:
add grid layout

Adds grid layout for home page

p-galligan committed:
change button styles

change styles for read more buttons

p-galligan committed:
remove old css files

Removes old css

p-galligan committed:
Update pages for grid layout

Updates internal pages with grid layout.

p-galligan committed:
update home page

Updates styles on the home page

p-galligan committed:
Tweak layouts for home and posts

Tweaks layout for home and post pages

p-galligan committed:
tweak post metadata

Tweaks styles on post metadata

daremiyo committed:
authors page link styling

daremiyo committed:
categories and tags links styling and colors

p-galligan commented:
Ok, so for sprint 2 we have:

1. Implemented skip links
2. Removed all local Sass/CSS files that were on the site
3. Implemented the Grid/main layout on all pages
4. Change/update button styles
5. Add delineation between posts on the home page
6. Change font and header styles
7. Add image layout controls in posts
8. Tweak styles for dates and names on posts
9. Update styles for padding and fonts on Authos/Tags/Categories pages

Outstanding work:

1. Update styles for tag and category links on post pages
2. Tweak padding and font choices on all pages
3. Implement media query breakpoints for more responsiveness

p-galligan committed:
Update styles

Fix link font size in excerpt.

Change grid span and size.

Make read more buttons smaller and transparent.

HaSistrunk commented on _layouts/home.html:
```diff
@@ -10,50 +10,51 @@
+        <div class="pagination">
+          {% if paginator.previous_page %}
+          <a href="{{ paginator.previous_page_path | relative_url | replace: '//', '/' }}" class="btn btn--lg btn--blue">
+            <i class="fa fa-chevron-left"></i>
+            {{ site.theme_settings.str_previous_page }}
+          </a>
+          {% endif %}
+          {% if paginator.next_page %}
+          <a href="{{ paginator.next_page_path | relative_url | replace: '//', '/' }}" class="btn btn--lg btn--blue ml-60">
+            {{ site.theme_settings.str_next_page }}
+            <i class="fa fa-chevron-right"></i>
```
Instead of using the `<i>` element here, I suggest following the [Style Guide pattern for adding icons to buttons](https://styles.rockarch.org/?path=/story/components-button--icons) using material icons. It looks like the material icons we want here are "[chevron_right](https://fonts.google.com/icons?selected=Material+Symbols+Outlined:chevron_right:FILL@0;wght@400;GRAD@0;opsz@24&icon.size=24&icon.color=%235f6368)" and "[chevron_left](https://fonts.google.com/icons?selected=Material+Symbols+Outlined:chevron_left:FILL@0;wght@400;GRAD@0;opsz@24&icon.size=24&icon.color=%235f6368)".

HaSistrunk commented on _includes/header.html:
```diff
@@ -1,46 +1,93 @@
+			  </li>
+			</ul>
+	
+			<div class="dropdown hide-on-lg-up">
+			  <button
+				id="nav-toggle"
+				class="btn btn--navy nav__btn--mobile closed"
+				aria-expanded="false"
+			  >
+				<span class="material-icon">menu</span>
+			  </button>
```
To enable the functionality of this button on small screens, we'll need to add a nav-toggle.js script. We use this functionality in multiple sites, so you can just copy the existing script from another site. [Here it is in aquila](https://github.com/RockefellerArchiveCenter/aquila/blob/base/assign_rights/static/js/nav-toggle.js).

HaSistrunk commented on _includes/header.html:
```diff
@@ -1,46 +1,93 @@
+				</a>
+			  </li>
+			  <li class="nav__item btn--navy">
+				<a class="nav__link" href="/categories/" id="categories">
```
From the Travis test fail message, it looks like this link should be `/categories`, not `/categories/`?

p-galligan committed:
Add styles

Fix pagination buttons and nav-toggle

p-galligan committed:
add spaces between tags and categories

Adds spacing between tags and categories on post pages

HaSistrunk committed:
use style library favicon

HaSistrunk committed:
correct reading room hours

HaSistrunk committed:
Adjust text size and margins, add undefined element styles

HaSistrunk committed:
remove extra headers and footers, update footer link style

HaSistrunk committed:
adjust layout to mirror rockarch.org and docs sites

HaSistrunk committed:
insert script to add RAC style classes to Markdown tables

HaSistrunk committed:
style img elements

HaSistrunk committed:
use rouge highlighter custom styles to match Docs Site for code

p-galligan committed:
minor layout tweaks

updates post layouts

HaSistrunk committed:
fix selector

p-galligan committed:
fix typo

Fixes typo in formatting

p-galligan committed:
Merge branch 'issue-252' of https://github.com/RockefellerArchiveCenter/bits_and_bytes into issue-252

HaSistrunk committed:
simplify and comment out larger font-size styles to facilitate review

daremiyo committed:
use media query for spacing between posts

HaSistrunk committed:
implement backstopjs for visual regression testing

daremiyo committed:
media query for excerpt spacing

HaSistrunk committed:
use new style library version 0.11.0

p-galligan committed:
Update media queries

Updates margins for certain media query sizes

daremiyo commented:
Latest changes:
- Adjustments in header to make responsive for small screens
- Button class styling for continue reading buttons to make them stand out more
- Added spacing between tags and categories in posts
- Typography changes to make site resemble RAC main website and Docs Site
- Removed unnecessary elements like extra headers and footers
- Added RAC style library classes for Markdown tables
- Added rouge highlighter custom styles for code like in Docs Site
- Layout adjustments in posts for author alignment and metadata
- Implemented media queries for responsive spacing between elements
- Added RAC Style library updates

HaSistrunk committed:
consolidate and simplify custom h2 styles

HaSistrunk committed:
separate css declarations for clarity

HaSistrunk committed:
update backstop reference images after css changes

HaSistrunk commented:
This is looking really good to me! I made a few small tweaks to reorder and combine some CSS declarations in `custom.css` just to simplify things. A couple thoughts:

- I think in some posts the text feels a little hard to read, but that's because we have some long posts with very few headings to break up the text - a good reminder for us when drafting these posts. I don't think making the text size bigger or changing other typographic styles is going to fix that.
- We may want to look more comprehensively at the scripts in here to either update and/or remove. But that's something that may make sense to do as part of a later planned project to create a RAC scripts library and implement.

And two action items:
1. It would be great if someone else could test the BackstopJS implementation by following the steps in the README and running `yarn backstop-test` just to make sure the tests are working and they pass (I tested, but good to have a second check).
2. It looks like we can get rid of some existing classes in the html that aren't being used now: `authors`, `tags` (we are only using `tag`), `author-post`, `meta`, `category-post`, `call-out`, `continue-reading`, and `tagged-post`. I'm assuming we'd like to keep some of the unused html "includes" files for features we may want the flexibility to implement in the future.

p-galligan committed:
Move post metadata

Moves post authors and dates into post title section

p-galligan committed:
Merge branch 'issue-252' of https://github.com/RockefellerArchiveCenter/bits_and_bytes into issue-252

HaSistrunk committed:
establish consistent margin between title bar and post content

HaSistrunk committed:
cleanup unused css classes

HaSistrunk committed:
set max-width for wide screens and improve css comments

helrond commented on _config.yml:
```diff
@@ -18,7 +18,7 @@ theme_settings:
   header_text_feature_image:
   footer_text: >
-   Powered by <a href="https://jekyllrb.com/">Jekyll</a> with <a href="https://github.com/rohanchandra/type-theme">Type Theme</a>
+   Powered by <a class="footer-primary__link" href="https://jekyllrb.com/">Jekyll</a> with <a class="footer-primary__link" href="https://github.com/rohanchandra/type-theme">Type Theme</a>
```
Given all the customization we've done I'm not sure this text is accurate.

helrond commented on _includes/header.html:
```diff
@@ -1,46 +1,93 @@
+	<div class="wrapper">
+	  <div class="container flex">
+		<div class="header__brand header__brand--text">
+		  <a href="/" class="header__brand-title">Bits & Bytes</a>
```
I'm seeing this misaligned on mobile (see screenshot) - is anyone else seeing this too?
<img width="499" alt="Screen Shot 2024-07-22 at 10 45 22 AM" src="https://github.com/user-attachments/assets/7c837fbe-2ae3-4af8-960b-801f2b5f2f61">

p-galligan replied:
Hey, I'm seeing this as well, but I'm not sure what's causing this.

Do you see anything @HaSistrunk

HaSistrunk replied:
Ah good catch. This is an issue in the Style Library, so also occurs in other site headers like the Docs site. Let me apply a fix in the Style Library and then we can roll it back into this site.

helrond reviewed (changes requested):
Two really small things, otherwise looks great to me!

p-galligan committed:
remove powered by type theme

Removes the powered by Type theme text from footer

HaSistrunk committed:
fix header layout issue with style library version update

HaSistrunk committed:
reference correct repo in readme

helrond commented on _config.yml:
```diff
@@ -18,7 +18,7 @@ theme_settings:
   header_text:
   header_text_feature_image:
   footer_text: >
```
It looks like this variable (`footer_text`) is no longer being used? If that's true, let's just delete it along with any other unused configs.

helrond reviewed (changes requested):
One last thing...

p-galligan committed:
Simplify config

Remove unused/unnecessary configs.

HaSistrunk committed:
update backstop images with header changes

HaSistrunk committed:
Merge branch 'gh-pages' into issue-252

p-galligan committed:
Add hero logo

Adds hero logo to the first page of the site

HaSistrunk committed:
adjust hero size, alignment, and text

HaSistrunk committed:
add alt text to logo

HaSistrunk committed:
finalize hero text and layout

HaSistrunk committed:
do not make hero an h1

HaSistrunk committed:
update social media icons in footer

HaSistrunk committed:
update backstop ref imgs
